core/src/costs.rs

```diff
@@ -39,501 +38,461 @@ impl<P: PageNumber> CostPerPage<P> {
         }
     }
 
-    /// Calculate cost for `pages`.
-    pub fn calc(&self, pages: P) -> u64 {
-        self.cost.saturating_mul(pages.raw() as u64)
-    }
-
-    /// Cost for one page.
+    /// Cost for one.
     pub const fn one(&self) -> u64 {
         self.cost
     }
+}
 
-    /// Returns another [CostPerPage] with increased `cost` to `other.cost`.
-    pub const fn saturating_add(&self, other: Self) -> Self {
-        Self::new(self.cost.saturating_add(other.cost))
+impl<T: Into<u32>> CostPer<T> {
+    /// Calculate cost for `num` amount of `T`.
+    pub fn calc(&self, num: T) -> u64 {
```

Comment:
what about just `calc_for` ? `saturating` - I think is redundant, `for_many` - it's not actually correct, because it also can calculate for one or for zero